_posts/2019-09-10-unbearable-lightness-of-refactoring.markdown

```diff
@@ -0,0 +1,115 @@
+---
+layout: epic
+title: "Unbearable Lightness of Refactoring"
+date: "2019-09-10"
+author: [ashkan]
+categories: [refactoring, community, culture]
+comment_id: 522
+---
+
+As engineers we are constantly in the process of building new features and improving our existing ones. Nowadays, with the help of tools and processes like code reviews one could argue the quality of the code being written has risen. At Artsy a pull request normally has one Assignee and possibly one or more Reviewers, so why we still do lot of refactoring?
+
+> There is no means of testing which decision is better, because there is no basis for comparison. We live everything as it comes, without warning, like an actor going on cold. And what can life be worth if the first rehearsal for life is life itself?
+
+― Milan Kundera, [The Unbearable Lightness of Being](https://en.wikipedia.org/wiki/The_Unbearable_Lightness_of_Being)
+
+Part of me wants to end this blogpost by Kundra’s quote, but for now lets get deeper.
+
+<!-- more -->
+
+## "The Refactor"
+Recently we've started adding SCA (Strong Customer Authentication) support to one of our services. This service is relatively young in our stack and very well reviewed. While the original code and approach looked nice and simple, as this service naturally grew and we started adding more logic to it, things got more and more complicated. During SCA support efforts, we realized its time to refactor. The code I was trying to refactor was less than a year old and originally written, well... by me!
```

Comment:
```suggestion
Recently we've started adding [SCA (Strong Customer Authentication)](https://stripe.com/docs/strong-customer-authentication) support to one of our services. This service is relatively young in our stack and very well reviewed. While the original code and approach looked nice and simple, as this service naturally grew and we started adding more logic to it, things got more and more complicated. During SCA support efforts, we realized its time to refactor. The code I was trying to refactor was less than a year old and originally written, well... by me!
```

Reply:
Up to you, but `originally written, well... by me!` could link to https://twitter.com/davidwalshblog/status/953663412013293569